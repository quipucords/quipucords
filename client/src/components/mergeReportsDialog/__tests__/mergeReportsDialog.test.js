import React from 'react';
import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import MergeReportsDialog from '../mergeReportsDialog';

describe('ToastNotificationsList Component', () => {
  const generateEmptyStore = () => configureMockStore()({ scans: {} });

  it('should shallow render a basic component', () => {
    const store = generateEmptyStore();
    const props = { show: true };
    const wrapper = shallow(<MergeReportsDialog {...props} show details={false} scans={[{ id: 1, name: 'test' }]} />, {
      context: { store }
    });

    expect(wrapper.dive()).toMatchSnapshot();
  });
});
