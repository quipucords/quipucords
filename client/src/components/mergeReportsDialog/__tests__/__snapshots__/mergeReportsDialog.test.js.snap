// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastNotificationsList Component should shallow render a basic component 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={false}
    closeLabel="Close"
  >
    <button
      aria-hidden="true"
      aria-label="Close"
      className="close"
      onClick={[Function]}
      type="button"
    >
      <Icon
        name="close"
        type="pf"
      />
    </button>
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    >
      Summary Merge Report
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <div
      className="merge-reports-body"
    >
      <span
        className="merge-reports-icon"
      >
        <Icon
          name="error-circle-o"
          type="pf"
        />
      </span>
      <span>
        <h3
          className="merge-reports-heading"
        >
          This action is invalid. You must select at least two scans with successful most recent scans.
        </h3>
        <div>
          <div>
            <span>
              Failed scans that cannot be included in the merged report:
            </span>
            <ul>
              <li
                key="1"
              >
                test
              </li>
            </ul>
          </div>
        </div>
      </span>
    </div>
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="primary"
      className="btn-cancel"
      disabled={false}
      onClick={[Function]}
    >
      Close
    </Button>
  </ModalFooter>
</Modal>
`;
